cpp/include/raft/neighbors/ivf_pq_helpers.cuh

```diff
@@ -138,6 +225,43 @@ void pack_list_data(raft::resources const& res,
   ivf_pq::detail::pack_list_data(res, index, codes, label, offset);
 }
 
+/**
+ * Write flat compressed PQ codes into an existing list by the given offset.
+ *
+ * The list is identified by its label.
+ *
+ * NB: no memory allocation happens here; the list must fit the data (offset + n_vec).
+ *
+ * Usage example:
+ * @code{.cpp}
+ *   // We will write into the 137th cluster
+ *   uint32_t label = 137;
+ *   // allocate the buffer for the input codes
+ *   auto codes = raft::make_device_matrix<const uint8_t>(res, n_vec, raft::ceildiv(index.pq_dim() *
+ * index.pq_bits(), 8));
+ *   ... prepare n_vecs to pack into the list in codes ...
+ *   // write codes into the list starting from the 42nd position
+ *   ivf_pq::helpers::pack_compressed_list_data(res, &index, codes.data_handle(), n_vec, label, 42);
```

Comment:
What is n_vec and nlabel? where do `res` and `index` get created? And how can we assume `index` needs to be passed by reference? Please add these details to these usage examples. It may seen redunant here, but it's important that users have these examples.

Reply:
`res` is just the raft handle. I was under the assumption that things like the raft resource can be considered self-explanatory.

Reply:
Copy/paste. I wouldn't assume anything is self explanatory to a user. It's self explanatory to us because we know well how raft works.